### PR TITLE
Empty strings are valid properties and other changes

### PR DESCRIPTION
**Pull request by acrrd**

### CODE REVIEW

**Review by joecks — APPROVED** [review]

**Review by joecks — APPROVED** [review]

File: .spectral.js [hunk 1]
```diff
@@ -50,6 +50,8 @@ module.exports = {
     },
     'ibm-major-version-in-path': 'off',
     'ibm-schema-description': 'off',
+    // spectral has is own rules to check schema that support v3.1
```
Comment (janpetschexain) [inline .spectral.js:53]: :art: typos
```suggestion
    // spectral has its own rules to check schemas that support v3.1
```

File: web-api/openapi/schemas/document.yml [hunk 2]
```diff
@@ -11,29 +11,32 @@ DocumentPropertyId:
...
+  items:
+    $ref: '#/DocumentPropertyString'
+
 DocumentProperty:
   description: | 
-    Arbitrary property that can be attached to a document.
-    When a property is set to `null`, it is treated as if that property had no values.
+    Arbitrary data that can be attached to a document. A subset of JSON is supported.
+    When a property is set to null, or an empty array it is treated as though that property has no values.
+    This means that they will not appear in searches with filter on this properties.
```
Comment (janpetschexain) [inline web-api/openapi/schemas/document.yml:32]: :art: phrasing
```suggestion
    When a property is set to null or an empty array, it is treated as if that property had no values.
    This means that they won't appear in searches with filters on these properties.
```

File: web-api/openapi/schemas/document.yml [hunk 3]
```diff
@@ -11,29 +11,32 @@ DocumentPropertyId:
 DocumentPropertyString:
   description: |
     Arbitrary string property that can be attached to a document.
-    Enclosing whitespace will be trimmed.
     The length constraints are in bytes, not characters.
   type: string
   pattern: '^[^\x00]+$'
-  minLength: 1
+  minLength: 0
```
Comment (janpetschexain) [inline web-api/openapi/schemas/document.yml:17]: ❓ why should we accept empty strings at all? if we get an empty string, this is most likely a logic error on the customer side. and it also doesn't bring any benefit to our filter logic, on the contrary.

❓ also, why shouldn't we trim them, or if we don't want to change the input, why don't we at least validate if they are trimmed?

File: web-api/openapi/schemas/document.yml (hunk 2 above)
Comment (acrrd) [reply web-api/openapi/schemas/document.yml]: I should trust less the phrasing from ES documentation :grimacing:

**Pull request merged**